src/ui/Speed.jsx

```diff
@@ -15,6 +19,37 @@ export function Speed() {
     }, 100)
     return () => clearInterval(interval)
   }, [])
+
+  useEffect(() => {
+    if (boost && nitroRef.current !== null) {
+      let level = useStore.getState().nitro.level
+      const interval = setConsumption(level, true)
+      return () => clearInterval(interval)
+    } else {
+      const interval = setConsumption(null, false)
+      return () => clearInterval(interval)
+    }
+  }, [boost])
+
+  const setConsumption = (level, indicator) => {
+    const interval = setInterval(() => {
+      if (indicator) {
+        level -= 10
+        if (level <= 300 && level >= 250) {
+          nitroRef.current.style.stroke = '#FFE600'
+        } else if (level < 250 && level >= 200) {
+          nitroRef.current.style.stroke = '#FF0000'
+          if (!nitroRef.current.classList.contains('nitro-path-blink')) nitroRef.current.classList.add('nitro-path-blink')
+        } else if (level <= 199) {
+          return
+        }
+        nitroRef.current.style.strokeDasharray = level
+        set((state) => ({ ...state, nitro: { level: level } }))
+      }
+    }, 100)
+    return interval
+  }
```

Comment:
This can be simplified a lot

Reply:
If there is a possibility to get the keydown event of boost or to set useFrame instead of useEffect then it could be done without setInterval. Any other suggestions?

Reply:
I suggest doing all of this in useFrame like the camera sway here: https://github.com/pmndrs/racing-game/blob/main/src/models/vehicle/Vehicle.jsx#L87-L93 but instead of mutating the camera you need to mutate a `boostLeft` prop

I would also suggest changing the NitroBar component to take in a prop `boostLeft` that styles the component instead of using a ref

Reply:
the boost in the store could change to:
```
boost: {
 activated: boolean,
 boostLeft: float
}
```

Reply:
Just a suggestions is to rename the boostLeft to remaingBoost or along those lines. I've interpreted 3x by now whereby I thought the Left was meant as in left/right/up/down way

Reply:
Also please normalize remainingBoost [0-100]